src/test/groovy/com/infusionsoft/gradle/version/GitVersionResolverTest.groovy

```diff
@@ -44,7 +52,7 @@ class GitVersionResolverTest extends RepositoryTestCase {
     @Test
     void testGetVersionExistingReleaseWithUncommittedChanges() {
         String tagVersion = '0.0.1'
-        String expectedVersion = tagVersion + defaultTagOptions.snapshotSuffix
+        String expectedVersion = '0.0.2' + defaultTagOptions.snapshotSuffix
```

Comment:
Understood.  I would argue that once you have released version x.y.z, if you have local changes, you shouldn't be going back to x.y.z-SNAPSHOT (since x.y.z has already been released), but you should be moving onto x.y.(z+1)-SNAPSHOT, since x.y.(z+1) is the next version that will be released, barring an explicit change to (x+1).0.0 or x.(y+1).0